### PR TITLE
Migrating Concourse CI pipelines to GH Actions

### LINKED ISSUE #1202: arm64 support
M1 has been around for a long time, Ampere Altra is taking over the server world. Builds under Arm64 would be extremely useful (the rabbitmq itself supports).

You can find an example of solving the assembly problem [here](https://github.com/egeneralov/cluster-operator/compare/main...egeneralov:cluster-operator:arm#diff-e426ed45842837026e10e66af23d9c7077e89eacbe6958ce7cb991130ad05ada)

### PR DESCRIPTION
**Pull request by coro**
This closes #1202 

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed

## Summary Of Changes

- Move the cluster operator to be a multiarch OCI built on AMD64 and ARM64, instead of just the former.
- Move existing Concourse CI infrastructure to GitHub Actions

## Additional Context

The pipeline now runs, on PRs:

- kubectl tests
- Unit tests
- Integration tests
- Tests that a multiarch OCI can be successfully built, but does not push it
- Runs system tests locally against kind, for multiple RabbitMQ Cluster versions
  - This excludes the volume expansion system tests
- Runs the documented examples against kind in 'dry-run' mode

On pushes to main:
- kubectl tests
- Unit tests
- Integration tests
- Builds a multiarch OCI of the operator (pushed to `rabbitmqoperator/cluster-operator` with a tag of the SHA256 and `edge`), and a manifest referencing that OCI
- Also builds a single-arch version of the image, for AMD64, for compatibility with internal tooling
- Runs system tests against GKE, for multiple RabbitMQ Cluster versions
- Runs the documented examples against GKE
- Tests if upgrading to the operator built earlier would cause a rolling restart of RabbitmqClusters

On creation of a tag in the repo:
- Same as push to main, except in addition:
  - OCI is tagged also with `latest` if the latest version, and the semver version of the release
  - Uploads manifests to a release in GitHub as a Draft release

N.B.: This PR removes the use of the rabbitmqoperator/cluster-operator-dev image, as it's now functionally replaced by the regular image with a SHA tag.

## Local Testing

Please ensure you run the unit, integration and system tests before approving the PR.

To run the unit and integration tests:

```
$ make unit-tests integration-tests
```

You will need to target a k8s cluster and have the operator deployed for running the system tests.

For example, for a Kubernetes context named `dev-bunny`:
```
$ kubectx dev-bunny
$ make destroy deploy-dev
# wait for operator to be deployed
$ make system-tests
```

### CODE REVIEW

**Comment by ablease** [comment]
Based upon sequential changes to the same file you could squash the following sets of commits. 

Makefile changes: [c8aa3a1884329fe9fc7394def3e7ac5553422255 b490b8ca7518b785209a2944df73e11b22a055e1]
Build-Test-Publish changes: [30e571b8b3c5c346e5369f78ef42972321301ce0 eb3ef546c8a40c1d90dc1ec5cce3aac915778326 8806354f78de94e0b76e1117a31b747c950614d0 6395e014c8e952536270cc92d553a44bb2b1dbbc 5b60d848194ab69edbbb80362ce264da9d885ccb 05c2ec9204bb6fa4ee9bb58178605d4f77a53faa 037c46ccaf4d9d5363944c5f1b7606091666a654 2338cddd5ed7ad97d167a4129645af21e0c4abc0]

Makefile Changes Round 2: [afb8bf2a696fc07bc42a192d54df5e84ea67aaf4 12013713f0ee59b9d8468064b9d3c8eb43dcf117]

Build-test-publish Round 2 [e7500377e3f6a58353246b80550dfdab810a5084 e7b1016b929ace55a27f61ae4491db96bc0e4e31]

Build-test-publish Round 3 [dbef9a369f28e935606b9063fb685ed5a5e70a9c  70a94282126e802c181f3f650b5350c62a708088]

Build-test-publish Final Round! [40ab6c1792aaefe30708e2d957637f37fd985a83 f06842dac1997bc07cd91eecbafc263c74b6ec53]


I can't say to much about the github actions changes themselves, will keep an eye on them as they run :) Thanks!

**Review by ChunyiLyu — APPROVED** [review]
Some minor comments. Looks good!

File: .github/workflows/build-test-publish.yml [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+          manifest_path="$manifest_path -f $a"
+        done
+        # The examples are assumed to run in the 'examples' namespace, create if it doesn't exist
+        kubectl create namespace examples --dry-run=client -o yaml | kubectl  apply -f -
+        kubectl --namespace=examples apply --dry-run=server $manifest_path
+  system_tests:
+    name: System tests
+    runs-on: ubuntu-latest
+    container: us.gcr.io/cf-rabbitmq-for-k8s-bunny/rabbitmq-for-kubernetes-ci
+    if: ${{ github.event_name != 'pull_request' }}
```
[the remark this answers is not in the document]
Comment (coro) [reply .github/workflows/build-test-publish.yml:210]: [These lines](https://github.com/rabbitmq/cluster-operator/blob/c76db136b918e892a34c0617fc93521ee8341695/.github/workflows/build-test-publish.yml#L3-L14) limit this workflow to only be triggered on push (either to main, or the creation of a new tag) or PR, so by saying 'not on PRs' we're effectively saying 'only on new tags or commits to main'.

File: Makefile [hunk 2]
```diff
@@ -9,8 +9,15 @@ help:
 ENVTEST_K8S_VERSION ?= 1.22.1
```
[the remark this answers is not in the document]
Comment (coro) [reply Makefile:9]: It's used for `setup-envtest`, which is run for unit tests and integration tests.

**Pull request merged**